Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM python:3.9.0-slim-buster as base
+
+EXPOSE 5000
+
+RUN apt-get update \
+    && apt-get install -y curl \
+    && curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+
+ENV PATH="${PATH}:/root/.poetry/bin"
+
+WORKDIR /app
+COPY . .
```

Comment:
You could copy just the poetry files and then copy the rest after poetry install.